src/zh/cartoon18/src/eu/kanade/tachiyomi/extension/zh/cartoon18/Cartoon18.kt

```diff
@@ -0,0 +1,230 @@
+package eu.kanade.tachiyomi.extension.zh.cartoon18
+
+import android.app.Application
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.net.URLDecoder
+
+class Cartoon18 : HttpSource(), ConfigurableSource {
+    override val name = "Cartoon18"
+    override val lang = "zh"
+    override val supportsLatest = true
+
+    override val baseUrl = "https://www.cartoon18.com"
+
+    private val baseUrlWithLang get() = if (useTrad) baseUrl else "$baseUrl/zh-hans"
+
+    override val client = network.client.newBuilder().followRedirects(false).build()
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", baseUrl)
+    }
```

Comment:
```suggestion
    override fun headersBuilder() = super.headersBuilder()
        .add("Referer", "$baseUrl/")
```
preserve useragent and trailing slash in the referer